settings.yml

```diff
@@ -1,41 +1,115 @@
 ---
 default:
+  # Default pipename for status controller, used when pipename is not provided.
   pipename: 'periodic-nightly'
+
+  # YAML file where admins emails are stored.
+  # Necessary for admin authorization. 
   users_file: 'users.yml'
+
+  # Flask attribute needed for secure signing session cookies.
+  # Read more: http://flask.pocoo.org/docs/1.0/config/#SECRET_KEY
   secret_key: b'\x8f\xa6gL\xc7\xc8\xe7\xf4v\xe8\xfc\xef\x85\xd3\xa6\x142\xc1\xc3\x12\xcc\xcfp['
-  openid_provider: 'https://login.launchpad.net/+openid'
-database:
-  provider: 'mysql'
-  host: '10.10.10.5'
-  user: 'root'
-  password: 'root'
-  port: 3306
-  name: 'zuul'
-buildset:
-  per_page: 20
-  page_links: 4
-  log_url: 'http://logs.opencontrail.org'
+
+
+# Configuration of authentication.
+authentication:
+  # Specify providers of open id authentication system.
+  # Every provider must have an url attribute.
+  openid_providers:
+    launchpad:
+      url: 'https://login.launchpad.net/+openid'
+
+
+# Configuration of history feature.
+history:
+  # <protocol>://<host> for server which store logs.
+  log_server: 'http://logs.opencontrail.org/'
+
+  pagination:
+    # Number of elements displayed on page.
+    per_page: 20
+
+    # Number of links displayed before or after current page link.
+    page_links: 4
+
+
 zuul:
-  url: 'http://10.10.10.5/'
+  # <protocol>://<host> for zuul server.
+  zuul_server: 'http://10.10.10.5/'
```

Comment:
Note sure about naming here. Looks weird in code when referred to:
For ex: zuul_url = current_app.config["zuul"]["zuul_server"]
Maybe leave it as 'url'. The format of the value is an url, so 'url' is still quite fitting.